scripts/core/sync_from_manifest.py

```
@@ -0,0 +1,745 @@
+        else:  # dir
+            tgt_dir = target_root / op.relpath
+            existing = _read_dir(tgt_dir) if tgt_dir.is_dir() else {}
+            for rel, content in op.dir_files.items():
+                full = f"{op.relpath}/{rel}"
+                if rel not in existing:
+                    added.append(full)
+                elif existing[rel].data != content.data:
+                    modified.append(full)
+            for rel in existing:
+                if rel not in op.dir_files:
+                    removed_files.append(f"{op.relpath}/{rel}")
```

Ruff (0.15.20)

[warning] Use `list.extend` to create a transformed list

(PERF401)